### PR TITLE
Feature/change mqtt to http

### PR DESCRIPTION
**Pull request by RebeccaBorgqvist**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

### CODE REVIEW

File: homeassistant/components/nanogrid_air/api.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    except socket.gaierror as e:
+        _LOGGER.error("Failed to resolve hostname: %s", e)
+        return
+
+
+async def fetch_mac():
+    """Fetch mac address from status API."""
+    ip = await get_ip()
+
+    url_status = f"http://{ip}/status/"
```
Comment (NotCosyMax) [inline homeassistant/components/nanogrid_air/api.py]: While it seems technically safe, checking the IP return (so that it is not null) before sending get request might be an idea? As it seems now, get_ip() can return blank in case of an exception and at this point there is no point sending out a request? :)

File: homeassistant/components/nanogrid_air/config_flow.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        mac_address = await fetch_mac()
+        if any(mac_address):
+            unique_id = mac_address
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+        else:
+            return self.async_abort(reason="not_responding")
+
+        return self.async_create_entry(title="Nanogrid Air", data={"url": self._url})
```
Comment (NotCosyMax) [inline homeassistant/components/nanogrid_air/config_flow.py]: Seems like there is no option for the user to manually assign a IP if needed?

File: homeassistant/components/nanogrid_air/api.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        except aiohttp.HttpProcessingError as exc:
+            _LOGGER.error(
+                "HTTP request error occurred: %s - %s", exc.status, exc.message
+            )
+            return {}
+
+
+async def fetch_meter_data():
+    """Fetch data from the API and return as a dictionary."""
+    ip = await get_ip()
```
Comment (NotCosyMax) [inline homeassistant/components/nanogrid_air/api.py]: Each data update results in the need for a hostname lookup. Seems we should be able to assume the resolved IP to remain and only perform a new lookup if for nay reason are unable to reach the device as expected?

File: homeassistant/components/nanogrid_air/config_flow.py (hunk 2 above)
Comment (RebeccaBorgqvist) [reply homeassistant/components/nanogrid_air/config_flow.py]: No we don't want the user to be given that option, the code itself is supposed to do all the work. The less the user need to do, the better, according to HA.

Comment (NotCosyMax) [reply homeassistant/components/nanogrid_air/config_flow.py]: What about case where mDNS is not allowed for reasons? Allowing a fallback does not mean we don't provide "single click configuration" as well :)

File: homeassistant/generated/config_flows.py
Comment (GaroDemergian) [inline homeassistant/generated/config_flows.py:1]: I am unsure if we should be pushing changing done to this file as it is within the generated folder. Please check in HA docs if this is correct.

File: homeassistant/generated/integrations.json
Comment (GaroDemergian) [inline homeassistant/generated/integrations.json:1]: same here

File: homeassistant/generated/zeroconf.py
Comment (GaroDemergian) [inline homeassistant/generated/zeroconf.py:1]: same here

File: homeassistant/components/nanogrid_air/manifest.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "name": "Nanogrid Air",
+  "codeowners": ["@chargestorm"],
+  "config_flow": true,
+  "documentation": "https://www.home-assistant.io/integrations/nanogrid_air",
+  "iot_class": "local_polling",
+  "zeroconf": [
+    {
+      "type": "_http._tcp.local.",
+      "name": "ctek-ng-air.local."
+    }
```
Comment (GaroDemergian) [inline homeassistant/components/nanogrid_air/manifest.json]: We are not using zeroconf anymore. we are resolving the ip using socket.

File: tests/components/nanogrid_air/conftest.py
Comment (GaroDemergian) [inline tests/components/nanogrid_air/conftest.py:1]: Unit tests are broken so they shouldn't be part of this PR

File: tests/components/nanogrid_air/test_config_flow.py
Comment (GaroDemergian) [inline tests/components/nanogrid_air/test_config_flow.py:1]: Same here

File: homeassistant/components/nanogrid_air/sensor.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    async def update_data():
+        meter_data = await fetch_meter_data()
+        return meter_data
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="nanogrid_air",
+        update_method=update_data,
+        update_interval=timedelta(seconds=1),
```
Comment (GaroDemergian) [inline homeassistant/components/nanogrid_air/sensor.py]: update interval of 1 second might be an overkill. make sure that NGA is updating its reading that frequently.

File: tests/components/nanogrid_air/conftest.py
Comment (RebeccaBorgqvist) [reply tests/components/nanogrid_air/conftest.py:1]: Tests will be deleted in this PR, but included in next one

File: tests/components/nanogrid_air/test_config_flow.py
Comment (RebeccaBorgqvist) [reply tests/components/nanogrid_air/test_config_flow.py:1]: Tests will be deleted in this PR, but included in next one

File: homeassistant/generated/config_flows.py
Comment (RebeccaBorgqvist) [reply homeassistant/generated/config_flows.py:1]: I'll refer to HA website: https://developers.home-assistant.io/docs/development_checklist

File: homeassistant/generated/integrations.json
Comment (RebeccaBorgqvist) [reply homeassistant/generated/integrations.json:1]: See previous answer

File: homeassistant/generated/zeroconf.py
Comment (RebeccaBorgqvist) [reply homeassistant/generated/zeroconf.py:1]: This is now deleted after we removed it from manifest.json.

File: homeassistant/components/nanogrid_air/manifest.json (hunk 4 above)
Comment (RebeccaBorgqvist) [reply homeassistant/components/nanogrid_air/manifest.json]: This is now removed

File: homeassistant/components/nanogrid_air/sensor.py (hunk 5 above)
Comment (RebeccaBorgqvist) [reply homeassistant/components/nanogrid_air/sensor.py]: Changed to 10sek update interval

File: homeassistant/components/nanogrid_air/config_flow.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_URL
+
+from .api import fetch_mac, get_ip
+from .const import DOMAIN
+
+API_DEFAULT = "http://ctek-ng-air.local/"
```
Comment (GaroDemergian) [inline homeassistant/components/nanogrid_air/config_flow.py]: should be "http://ctek-ng-air.local/meter/"

File: homeassistant/components/nanogrid_air/config_flow.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+IP_DEFAULT = "Delete me and enter device's IP"
```
Comment (GaroDemergian) [inline homeassistant/components/nanogrid_air/config_flow.py]: Change to IP_INPUT_FIELD instead of IP_DEFAULT

File: homeassistant/generated/config_flows.py
Comment (GaroDemergian) [inline homeassistant/generated/config_flows.py:1]: Still not removed

File: homeassistant/generated/integrations.json
Comment (GaroDemergian) [inline homeassistant/generated/integrations.json:1]: Still not removed

File: homeassistant/components/nanogrid_air/config_flow.py (hunk 7 above)
Comment (RebeccaBorgqvist) [reply homeassistant/components/nanogrid_air/config_flow.py]: i'll fix that

File: homeassistant/components/nanogrid_air/config_flow.py (hunk 6 above)
Comment (RebeccaBorgqvist) [reply homeassistant/components/nanogrid_air/config_flow.py]: fixed

File: homeassistant/components/nanogrid_air/api.py (hunk 1 above)
Comment (RebeccaBorgqvist) [reply homeassistant/components/nanogrid_air/api.py]: Fixed

File: homeassistant/components/nanogrid_air/config_flow.py (hunk 2 above)
Comment (RebeccaBorgqvist) [reply homeassistant/components/nanogrid_air/config_flow.py]: Fixed

File: homeassistant/components/nanogrid_air/api.py (hunk 3 above)
Comment (RebeccaBorgqvist) [reply homeassistant/components/nanogrid_air/api.py]: Fixed

**Review by GaroDemergian — APPROVED** [review]

**Review by NotCosyMax — APPROVED** [review]
Looks good to me... FOR NOW!

**Review by Wesztman — APPROVED** [review]
